meta-cli/src/cli/gen.rs

```diff
@@ -0,0 +1,173 @@
+        let config = Arc::new(dbg!(config));
+        let Some(mgen_conf) = &config.metagen else {
+            anyhow::bail!(
+                "no metagen section defined in config found at {:?}",
+                config.path
+            );
+        };
+        let node_config = config.node(&self.node, self.target.as_deref().unwrap_or("dev"));
+        let node = node_config
+            .build(&dir)
+            .await
+            .with_context(|| format!("building node from config: {node_config:#?}"))?;
+
+        let resolver = MetagenCtx {
+            config: config.clone(),
+            dir,
+            typegate: Arc::new(node),
+        };
+
+        let files = metagen::generate_target(mgen_conf, &self.gen_target, resolver).await?;
+        let mut set = tokio::task::JoinSet::new();
+        for (path, file) in files {
+            set.spawn(async move {
+                tokio::fs::create_dir_all(path.parent().unwrap()).await?;
+                tokio::fs::write(path, file).await?;
+                Ok::<_, tokio::io::Error>(())
+            });
+        }
+        while let Some(res) = set.join_next().await {
+            res??;
+        }
+
+        server_handle.unwrap().stop(true).await;
+
+        Ok(())
+    }
+}
+
+#[derive(Clone)]
+struct MetagenCtx {
+    config: Arc<Config>,
+    typegate: Arc<common::node::Node>,
+    dir: PathBuf,
+}
+
+impl InputResolver for MetagenCtx {
+    async fn resolve(&self, order: GeneratorInputOrder) -> anyhow::Result<GeneratorInputResolved> {
+        Ok(match order {
+            GeneratorInputOrder::TypegraphFromTypegate { name } => {
+                GeneratorInputResolved::TypegraphFromTypegate {
+                    raw: self
+                        .typegate
+                        .typegraph(&name)
+                        .await?
+                        .with_context(|| format!("no typegraph found under \"{name}\""))?,
+                }
+            }
+            GeneratorInputOrder::TypegraphFromPath { path, name } => {
+                let (tx, rx) = tokio::sync::oneshot::channel();
+                let config = self.config.clone();
+                let dir = self.dir.join(path);
+                tokio::task::spawn_blocking(move || {
+                    actix::run(async move {
+                        let res = load_tg_at(config, dir, name.as_deref()).await;
+                        tx.send(res).unwrap();
+                    })
+                })
+                .await??;
+                let raw = rx.await??;
+                GeneratorInputResolved::TypegraphFromTypegate { raw }
+            }
+        })
+    }
+}
+
+async fn load_tg_at(
+    config: Arc<Config>,
+    path: PathBuf,
+    name: Option<&str>,
+) -> anyhow::Result<Typegraph> {
+    ServerStore::with(
+        Some(crate::com::store::Command::Serialize),
+        Some(config.as_ref().clone()),
+    );
+    // ServerStore::set_prefix(self.prefix.to_owned());
+
```

Comment:
btw you can now call `ServerStore::set_artifact_resolution_flag(false)`, this will disable resolving a non existing mdk wasm file.